docs/decisions/0007-release-process.md

```diff
@@ -0,0 +1,334 @@
+   git push upstream s3gw-v0.11
+   ```
+
+9. At this point we will need to update the Helm Chart to reflect the release
+   version. This becomes bit tricky, because we want the change to be reflected
+   in both the `main` branch and the `s3gw-vX.Y` branch on the `s3gw-charts`
+   repository. To do this, we will apply a patch to the `main` branch, and then
+   backport the change to the release branch.
+
+   First, for `v0.11.0`, the chart version needs to be updated with the specific
+   version, at `charts/s3gw/Chart.yaml`. This change should then be committed,
+   and a Pull Request of this change opened against `main`.
+
+   Once the Pull Request has been merged, note down the new commit's `SHA`;
+   running `git log` should give you its value. We can now change to the
+   `s3gw-v0.11` branch, and `cherry-pick` the commit, and finally tag the branch
+   with the release version.
+
+   ```shell
+   git cherry-pick -x -s -S <SHA>
+   git tag --annotate --sign -m "Release v0.11.0" s3gw-v0.11.0
+   git push upstream s3gw-v0.11
+   ```
+
+10. With all sub-project repositories ready to be released, it's time to prepare
+    the `s3gw` repository for a release. Much like what we did for the
+    `s3gw-charts` repository, we will have to apply a patch on the `main` branch
+    first, and then backport it to the release branch: this time to keep track
+    of the `CHANGELOG`.
+
+    First step, we need to go to the [Current CHANGELOG][8] page on the
+    repository's Wiki, and copy the contents for the release version being
+    handled to a file at `docs/release-notes/s3gw-vX.Y.Z.md`. Keep in mind the
+    release notes should be easily consumable by a human. Feel free to take
+    inspiration on previous release notes, and maintain consistency with them.
+
+    Creating a Pull Request against `s3gw`'s `main` branch is the next step.
+    Once that has been merged, note down the new commit's `SHA`, go back to the
+    `s3gw-vX.Y` branch, and cherry-pick the commit, tagging the branch with for
+    our specific release.
+
+    ```shell
+    git cherry-pick -x -s -S <SHA>
+    git tag --annotate --sign -m "Release v0.11.0" s3gw-v0.11.0
+    git push upstream s3gw-v0.11
+    ```
+
+    By pushing the branch with the release tag, we will trigger the release
+    workflow that will build the various release artifacts and publish the
+    containers on Quay.
+
+11. The last step is to officially release the new version. During the release
+    workflow, a [release draft][9] will be created. Once the release artifacts
+    have finished building, and have been published on Quay, we can then copy
+    the contents of the release notes file we created in step `10.`, and make
+    the release draft public.
+
+    It is advised that before making the release draft public, the list in
+    [Sanity Checks](#sanity-checks) be ensured to hold true.
+
+    With the release now complete, it is time to shout about it from the
+    rooftops.
+
```

Comment:
This is missing step a step 12 here: Sending out release announcements. This applies to:

rancher-users slack: #s3gw
SUSE slack: #discuss-s3gw
email: s3gw@suse.com

With growing stability of the project, other channels will probably need to be included here too.
We also might want to persist the template for these announcements somewhere